src/modules/localip.rs

```diff
@@ -0,0 +1,128 @@
+use super::{Context, Module};
+
+use crate::config::RootModuleConfig;
+use crate::configs::localip::LocalipConfig;
+use crate::formatter::StringFormatter;
+
+/// Creates a module with the ipv4 address of the local machine.
+///
+/// The `local_ipaddress` crate is used to determine the local IP address of your machine.
+/// An accurate and fast way, especially if there are multiple IP addresses available,
+/// is to connect a UDP socket and then reading its local endpoint.
+///
+/// Will display the ip if all of the following criteria are met:
+///     - localip.disabled is absent or false
+///     - localip.ssh_only is false OR the user is currently connected as an SSH session (`$SSH_CONNECTION`)
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let mut module = context.new_module("localip");
+    let config: LocalipConfig = LocalipConfig::try_load(module.config);
+
+    let ssh_connection = context.get_env("SSH_CONNECTION");
+    if config.ssh_only && ssh_connection.is_none() {
+        return None;
+    }
+
+    let localip = local_ipaddress::get().unwrap();
```

Comment:
This panics for me as soon as I disconnect my internet.

```
thread '<unnamed>' panicked at 'called `Option::unwrap()` on a `None` value', src/modules/localip.rs:25:42
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Reply:
Would unwrap_or_default() do the trick?

Reply:
Yep.